Create Trip Functionality

LLadas0805 opened the pull request:
# 🛠️ Pull Request Title

## 📋 Summary
This PR modifies the trip route backend slightly by modifying the document information that gets sent over to our database to resemble something closer to what the form sends to the route, and also the schema. I will eventually modify this more but for now I didn't want to change too much here yet since it works.

## ✅ What's Changed
- [ ] New features added?
- [ ] UI or style changes?
- [ ] Bug fixes?
- [x] Refactors or cleanup?

## 📎 Related Issues
_Closes #[issue number], or mention any relevant discussions._

## 🔍 How to Test & Screenshots
To start testing the trip creation feature, when you are on the user dashboard you can click the "create new trip" button.
<img width="1553" height="630" alt="image" src="https://github.com/user-attachments/assets/c8d96e0e-c375-4782-8371-b470d0edf954" />
Which will then bring you to this form and essentially allow you to create any trip you want with these inputs. I want to kind of see if there are any problems so far with error checking with the input but the actual trips themselves should be stored properly in Mongo.
<img width="2555" height="1264" alt="Screenshot 2025-07-20 205937" src="https://github.com/user-attachments/assets/b22bb33e-a918-4df0-b1e7-42e326014fbb" />
Finally when you create a trip, it should redirect you back to the dashboard and be seen in your trips list!
<img width="2554" height="1263" alt="Screenshot 2025-07-20 205959" src="https://github.com/user-attachments/assets/1d0d4c2a-194b-4c8e-8e29-478d8f2f1187" />


## 🙏 Reviewer Checklist
- [ ] Code is readable and commented
- [ ] No console errors or warnings
- [ ] UI works on desktop and mobile
- [ ] All functionality behaves as expected

LLadas0805 committed:
Modified schema and trip creation input

mjallen729 commented:
Looks good so far. There's some sort of issue with the line endings, not a huge deal but it's falsely flagging merge conflicts & showing weird change previews because of the whitespace diff. Could lead to some headaches in the future.

What OS are you developing on?

mjallen729 committed:
Fix whitespace issue

mjallen729 committed:
Merge branch 'Develop' into users/lladas/create-trip

mjallen729 commented:
Ok, was able to fix the line-ending issue by doing the following in terminal:

`
git config --global core.autocrlf input
`

Here's a summary of what that setting will do:
> When set to "input", Git will convert CRLF line endings (common in Windows) to LF (common in Unix-like systems) when committing files, but it will not convert them back to CRLF when checking out those files. This behavior is often used on Unix-like systems to ensure that files in the repository are consistently stored with LF line endings while still allowing Windows users to work with their preferred CRLF line endings locally.

Once you set the autocrlf config option, we shouldn't see this issue again. However, if it does persist you can run:
`
git add --renormalize .
`
`
git commit -m "Fix whitespace"
`

mjallen729 committed:
merge conflict pt 2

The addition logger utility was wrapped up in the merge conflict as well, I thought it was just the whitespace.

At this point the new logger changes are successfully incorporated into this branch without modifying any of the changes you made, Luke. Sorry for the confusion!

LLadas0805 reviewed (commented):
Thank you for incorporating the logger into the back-end!
